pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -311,10 +314,21 @@ func (plugin *cniNetworkPlugin) addToNetwork(network *cniNetwork, podName string
 		return nil, err
 	}
 
+	if res == nil {
+		glog.Warningf("CNI plugin gives empty network result")
+		return nil, nil
+	}
+
+	curRes, err := cnicurrent.NewResultFromResult(res)
+	if curRes != nil && len(curRes.IPs) != 0 {
+		glog.V(4).Infof("IP address for CNI network was set from plugin's output", netConf.Name, netConf.Plugins[0].Network.Type)
+		plugin.SetPodIPs(podSandboxID, curRes.IPs[0].Address.IP.To4().String())
```

Comment:
A couple of issues with this block, which is too simple:

- The first IP address in the list might be for any interface, not just the pod's interface.  CNI 0.3.0 and later plugins report multiple interfaces and IP addresses, and you need to make sure the IP is actually for the pod interface and not for say the host's bridge interface.
- The code isn't IPv6 compatible due to the To4()
- It should probably warn if the pod interface it finds is not called "eth0", which is what dockershim explicitly requests and which plugins are required to honor by the CNI spec

Instead, something like this:
```
// Find the pod's IP (ignoring any non-sandbox interfaces)
var podIP string
for _, ip := range curRes.IPs {
	if ip.Interface != nil {
		if len(curRes.Interfaces) > *ip.Interface {
			intf := curRes.Interfaces[*ip.Interface]
			if intf.Sandbox != "" && intf.Name == network.DefaultInterfaceName {
				// Found our sandbox interface, use the IP
				podIP = ip.Address.IP.String()
				break
			}
		}
	} else if podIP == "" {
		// Fall back to first un-associated address for compatibility
		// with older CNI plugins
		podIP = ip.Address.IP.String()
	}
}
```